tests/detectors/optimizations/constant_gtxn.py

```diff
@@ -0,0 +1,9 @@
+from tealer.detectors.optimizations.constant_gtxn import ConstantGtxn
+
+GTXN = """
+#pragma version 3
+int 1
+gtxns CloseRemainderTo
+"""
+
+constant_gtxn = [(GTXN, ConstantGtxn, [2, 3])]
```

Comment:
If could be done quickly, we should probably add a test in pyteal. It will also be useful for other constant detectors (gaids, gloads).

Reply:
We can probably use the source mapping for writing the test.

If source mapping is enabled, every line of the compiled teal code will contain the python source line. The test could be
```
def test():
    t1 = Assert(Gtxn[Int(1)].rekey_to() == Global.zero_address()) # 1: unoptimized gtxns

    receiver = Gtxn[Int(1)].receiver() # 2: unoptimized gtxns

    id = GeneratedID(Int(0)) # 1: unoptimized gaids
    [...]
    return Seq([
        t1,
        receiver,
        [...]
    ])


def compile() -> str:
    return compileTeal(test(), mode=Mode.Application, version=7)
```

```py
def test_detectors(...):
    teal = compile()
    # assert same number of instances are reported and every reported instance contains `unoptimized gtxns`.
    # also check there are no duplicate outputs.
```